synapse/cores/common.py

```diff
@@ -719,6 +719,21 @@ def getJoinByProp(self, prop, valu=None, mintime=None, maxtime=None, limit=None)
         '''
         return tuple(self._getJoinByProp(prop, valu=valu, mintime=mintime, maxtime=maxtime, limit=limit))
 
+    def getJoinInProp(self, prop, values, mintime=None, maxtime=None, limit=None):
```

Comment:
Having recently implemented 'by' syntax for swarm (https://github.com/vivisect/synapse/pull/30), it makes total sense to me, but I worry that any increased complexity is going to obfuscate the 'qbuild' query cache further. How would you feel about making query building just in time, but memoized?